closed/autoconf/custom-hook.m4

```diff
@@ -340,57 +339,86 @@ AC_DEFUN([OPENJ9_PLATFORM_EXTRACT_VARS_FROM_CPU],
-  AC_ARG_ENABLE([crac-support], [AS_HELP_STRING([--enable-crac-support], [enable CRAC support @<:@platform dependent@:>@])])
-  OPENJ9_ENABLE_CRAC_SUPPORT=false
+  AC_ARG_ENABLE([crac-support], [AS_HELP_STRING([--enable-crac-support], [enable CRaC support @<:@platform dependent@:>@])])
+  AC_ARG_ENABLE([criu-support], [AS_HELP_STRING([--enable-criu-support], [enable CRIU support @<:@platform dependent@:>@])])
+
+  # Complain about explicitly requested, but illegal combinations.
+  if test "x$enable_crac_support" = xyes && test "x$enable_criu_support" = xno ; then
+    AC_MSG_ERROR([--enable-crac-support requires CRIU support])
+  fi
+
+  # Compute platform-specic defaults.
+  case "$OPENJ9_PLATFORM_CODE" in
+    xa64)
+      default_crac=yes
+      default_criu=yes
+      ;;
+    xl64|xr64|xz64)
+      default_crac=no
+      default_criu=yes
+      ;;
+    *)
+      default_crac=no
+      default_criu=no
+      ;;
+  esac
 
+  # Capture the origin of each setting.
   if test "x$enable_crac_support" = xyes ; then
-    AC_MSG_RESULT([yes (explicitly enabled)])
-    OPENJ9_ENABLE_CRAC_SUPPORT=true
+    origin_crac="explicitly enabled"
   elif test "x$enable_crac_support" = xno ; then
-    AC_MSG_RESULT([no (explicitly disabled)])
+    origin_crac="explicitly disabled"
   elif test "x$enable_crac_support" = x ; then
-    case "$OPENJ9_PLATFORM_CODE" in
-      xa64)
-        AC_MSG_RESULT([yes (default)])
-        OPENJ9_ENABLE_CRAC_SUPPORT=true
-        ;;
-      *)
-        AC_MSG_RESULT([no (default)])
-        ;;
-    esac
+    # Adjust if CRUI is explicitly disabled.
+    if test "x$enable_criu_support" = xno && test "x$default_crac" = xyes ; then
+      origin_crac="implicitly disabled"
+      enable_crac_support=no
+    else
+      origin_crac=default
+      enable_crac_support=$default_crac
+    fi
   else
     AC_MSG_ERROR([--enable-crac-support accepts no argument])
   fi
-  AC_SUBST(OPENJ9_ENABLE_CRAC_SUPPORT)
-])
-
-AC_DEFUN([OPENJ9_CONFIGURE_CRIU_SUPPORT],
-[
-  AC_MSG_CHECKING([for CRIU support])
-  AC_ARG_ENABLE([criu-support], [AS_HELP_STRING([--enable-criu-support], [enable CRIU support @<:@platform dependent@:>@])])
-  OPENJ9_ENABLE_CRIU_SUPPORT=false
 
   if test "x$enable_criu_support" = xyes ; then
-    AC_MSG_RESULT([yes (explicitly enabled)])
-    OPENJ9_ENABLE_CRIU_SUPPORT=true
+    origin_criu="explicitly enabled"
   elif test "x$enable_criu_support" = xno ; then
-    AC_MSG_RESULT([no (explicitly disabled)])
+    origin_criu="explicitly disabled"
   elif test "x$enable_criu_support" = x ; then
-    case "$OPENJ9_PLATFORM_CODE" in
-      xa64|xl64|xr64|xz64)
-        AC_MSG_RESULT([yes (default)])
-        OPENJ9_ENABLE_CRIU_SUPPORT=true
-        ;;
-      *)
-        AC_MSG_RESULT([no (default)])
-        ;;
-    esac
+    # Adjust if CRaC is explicitly enabled.
+    if test "x$enable_crac_support" = xyes && test "x$default_criu" = xno ; then
+      origin_criu="implicitly enabled"
+      enable_criu_support=yes
```

Comment:
If `$default_criu` is `no`, this means `CRIU` is not supported in this platform, and can't be enabled.
An error should be thrown here.

Reply:
Just because the default is `no`, doesn't mean it's not supported. I think this will simplify things for someone working towards adding support on a platform where it is not enabled by default.

Reply:
It is my understanding that this is what you expected; see https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/800#issuecomment-2150302245.

Reply:
The script in question will enable `CRIU` implicitly when a user attempts to enable `CRaC` at a platform where `CRIU` is disabled by default.
Apparently, the user is working on `CRaC`, if the intention is to add `CRIU` support on that platform at the same time, I feel `CRIU` is better to be enabled explicitly, otherwise the compilation is going to fail.
In addition, I don't see any immediate possibility to add CRIU support in those default `no` platforms such as Windows, Mac, zOS, and AIX.

> It is my understanding that this is what you expected; see https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/800#issuecomment-2150302245.

Sorry I have to change it to:
* `--enable-crac-support` enabled `CRaC` where `CRIU` is enabled by default, otherwise an error will be thrown

Reply:
Again, a _default_ should not be equated with _support_. Before this change, you could use the options an all combinations, including one that makes no sense (given CRaC depends upon CRIU), or on platforms where it was unlikely to work. The aim of this change was to detect that one non-sensical choice early, saving time for a developer. I (and Tobi) agreed to your suggestions to allow `--enable-crac-support` to imply `--enable-criu-support`, and to have `--disable-criu-support` to imply `--disable-crac-support` making those outcomes easier to obtain.

I do not think this is the time (I'm not yet convinced it will ever be) to start introducing higher-level checking of configuration choices like you're now proposing. If you disagree, I suggest that is well outside the scope of this change.

Reply:
The configuration scenario in question is in the development stage, `origin_criu="implicitly enabled"` will show a hint of why a compilation occurs. 
I won't block the current change.